machine/parser.go

```diff
@@ -152,7 +152,7 @@ func unmarshallState(name string, raw_json *json.RawMessage) ([]state.State, err
 		passState := state.PassState{
 			Type:       to.Strp("Pass"),
 			ResultPath: path,
-			Result:     name,
```

Comment:
This is incorrect.

```
A: { ..., Next: X},
X: { Type: TaskFn, ...},
```

will exapnd to 

```
A: { ..., Next: X},
X: {Type: Pass, ResultPath: "$.Task", Result: "X", Next _X_}
_X_: { Type: Task, ...},
```

The Handler in Step is then the name of `$.Task == X` not `_X_`. The TaskFnName is kind of hidden from the user of step, so that when building stuff you don't need to worry about the `_X_` syntax at all.

A good way to check if your changes will break is to also `go get coinbase/odin`, don't use `dep ensure` to vendor `step` (this will use your local changes), then run Odin's tests.